File: Easy/1. Two Sum.py
class Solution:
    def twoSum(self, nums: list, target: int) -> list:
        numsMap = {num:index for index,num in enumerate(nums)}
        
        for index,num in enumerate(nums):
            if target-num in numsMap and numsMap[target-num] != index:
                return [index, numsMap[target-num]]

File: Easy/1. Two Sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int,int> map;
		vector<int> ans;
		
		for(int i = 0;i < nums.size(); i++){
			int num = nums[i];
			map[num] = i;
		}
		
		for(int i = 0;i < nums.size(); i++){
			int num = nums[i];
			if(map.count(target-num) > 0 && map[target-num] != i){
				ans.push_back(i);
                ans.push_back(map[target-num]);
				return ans;
			}
		}
        return ans;
    }
};